pkg/master/master.go

```diff
@@ -165,7 +165,14 @@ type thirdPartyEntry struct {
 type RESTOptionsGetter func(resource unversioned.GroupResource) generic.RESTOptions
 
 type RESTStorageProvider interface {
-	NewRESTStorage(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool)
+	NewRESTStorage(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) (groupInfo genericapiserver.APIGroupInfo, enabled bool)
+}
+
+// RESTStoragePostStartHookProvider is an optional interface in addition to a RESTStorageProvider that can provide a post start hook
+// for the api server
+type RESTStoragePostStartHookProvider interface {
+	RESTStorageProvider
+	PostStartHook(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) func() error
```

Comment:
> It's very weird to give these parameters to a post start hook. Maybe you should decide that the RESTStorage object should implement the hook, not the RESTStorageProvider.

I tried that.  I didn't want to force them all to implement when many won't need it.  I guess I'll keep a separate interface and build the hook during the `InstallAPI` step.